### PR TITLE
[TACHYON-1533] "bin/tachyon-mount.sh SudoMount workers" does not set the $bin variable

### PR DESCRIPTION
**Pull request by caadecarvalho**

### CODE REVIEW

File: bin/tachyon-workers.sh [hunk 1]
```diff
@@ -15,7 +15,7 @@ if [ $# -le 0 ]; then
 fi
 
 bin=`cd "$( dirname "$0" )"; pwd`
-DEFAULT_LIBEXEC_DIR="$bin"/../libexec
+DEFAULT_LIBEXEC_DIR=`echo $bin | sed "s/bin$/libexec/g"`
 TACHYON_LIBEXEC_DIR=${TACHYON_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
 . $TACHYON_LIBEXEC_DIR/tachyon-config.sh
```
Comment (caadecarvalho) [inline bin/tachyon-workers.sh]: If you guys prefer, I can make it one command, removing $bin variable as below:

DEFAULT_LIBEXEC_DIR=`cd "$( dirname "$0" )"; pwd | sed "s/bin$/libexec/g"`

Regards.

**Comment by caadecarvalho** [comment]
Hi guys, I left some comments for you. 
Regards.

**Comment by caadecarvalho** [comment]
I think should be better to change $DEFAULT_LIBEXEC_DIR variable inside all bin/tachyon-* files, can I make this change under this PR?

File: bin/tachyon-workers.sh [hunk 2]
```diff
@@ -15,7 +15,7 @@ if [ $# -le 0 ]; then
 fi
 
 bin=`cd "$( dirname "$0" )"; pwd`
-DEFAULT_LIBEXEC_DIR="$bin"/../libexec
```
Comment (uronce-cc) [inline bin/tachyon-workers.sh]: I think the original behavior is fine? What's the problem?

Comment (caadecarvalho) [inline bin/tachyon-workers.sh]: I was mentioned by Jiri Simsa that: '"bin/tachyon-mount.sh SudoMount
workers" does not set the $bin variable, resulting in incorrect behavior'
so I'm just giving a better assignment to those variables



Abraços,


Carlos Anibal Amaral de Carvalho
móvel: (11) 95203-1505
email: caadecarvalho@gmail.com
Facebook <https://www.facebook.com/camaralc> - LinkedIn
<http://br.linkedin.com/pub/carlos-anibal-amaral-de-carvalho/24/628/740/> -
Blog <http://caadecarvalho.wordpress.com>

2016-01-18 21:06 GMT-02:00 chang cheng <notifications@github.com>:

Comment (uronce-cc) [inline bin/tachyon-workers.sh]: @jsimsa Could you specify the problem you encountered? Thanks.

Comment (jsimsa) [inline bin/tachyon-workers.sh]: as @caadecarvalho mentioned, when you run `${TACHYON_HOME}/bin/tachyon-mount.sh SudoMount workers`, the `$bin` environment variable is never set and so the script does not do the right thing

Comment (caadecarvalho) [inline bin/tachyon-workers.sh]: @jsimsa, bin/tachyon-mount.sh sets $bin environment variable under function init_env when *mount_local* or *workers* 2nd parameter is passed , my thought was that you want to make a better variable assignment removing the two dots

Comment (jsimsa) [inline bin/tachyon-workers.sh]: the second parameter can be either "", "local", or "worker". When it is "workers", the init_env function is not invoked si the variable is not set.

Comment (caadecarvalho) [inline bin/tachyon-workers.sh]: it is invoked under *bin/tachyon-workers.sh* which loads env variables from *conf/tachyon-conf.sh*

Comment (uronce-cc) [inline bin/tachyon-workers.sh]: Hmm.. Doesn't [this line](https://github.com/amplab/tachyon/pull/2501/files#diff-40c38fb954cae1e4d8ca66ae6f3d7a52L17) set `bin`?

Comment (jsimsa) [inline bin/tachyon-workers.sh]: but bin/tachyon-workers.sh is never executed because $bin/tachyon-workers.sh is not resolved properly

Comment (jsimsa) [inline bin/tachyon-workers.sh]: see line 216

Comment (caadecarvalho) [inline bin/tachyon-workers.sh]: @jsimsa , yes you are right! Would be better to we execute a *init_env $0* before *$LAUNCHER ..*

Comment (jsimsa) [inline bin/tachyon-workers.sh]: it would be better not to use shell :-)

Comment (jsimsa) [inline bin/tachyon-workers.sh]: but, yes what you are proposing is the fix I was hoping for

Comment (caadecarvalho) [inline bin/tachyon-workers.sh]: hahaha, ok I work it out. Also, are you fine with those variable attributions? Eg, *$SOMEDIR/bin/../libexec*

**Comment by caadecarvalho** [comment]
Jenkins, test this please.

File: bin/tachyon-mount.sh [hunk 3]
```diff
@@ -213,6 +213,7 @@ case "${1}" in
         mount_local $1
         ;;
       workers)
+        bin=`cd "$( dirname "$0" )"; pwd`
         $LAUNCHER $bin/tachyon-workers.sh $bin/tachyon-mount.sh $1
```
Comment (caadecarvalho) [inline bin/tachyon-mount.sh]: Hi @jsimsa . Now, I just changed the *bin/tachyon-mount.sh* file, that should address your need.

Comment (uronce-cc) [inline bin/tachyon-mount.sh]: I feel better to call `init_env`

Comment (caadecarvalho) [inline bin/tachyon-mount.sh]: Sure @uronce-cc , the different thing on `init_env` it is that it have some variable definitions and a sanity check on memory worker which this is good for local worker but with remote workers, they will inherit $TACHYON_WORKER_MEMORY_SIZE from their local `libexec/tachyon-config.sh` . Was my understanding right?

Comment (uronce-cc) [inline bin/tachyon-mount.sh]: yes.

Comment (caadecarvalho) [inline bin/tachyon-mount.sh]: @uronce-cc , thanks, so I'll leave the code without the `init_env` function.

**Comment by caadecarvalho** [comment]
Jenkins, test this please.

File: bin/tachyon-mount.sh (hunk 3 above)
Comment (jsimsa) [inline bin/tachyon-mount.sh]: here is my suggestion,

move `bin=$(cd "$( dirname "$1" )"; pwd)` outside of init_env right after the LAUNCHER definition:

```
LAUNCHER=
# If debugging is enabled propagate that through to sub-shells
if [[ "$-" == *x* ]]; then
  LAUNCHER="bash -x"
fi
BIN=$(cd "$( dirname "$0" )"; pwd)
```

and replace occurrences of `$bin` with `${BIN}`

Comment (caadecarvalho) [inline bin/tachyon-mount.sh]: Nice @jsimsa , there are many references to `$bin` in other `bin/tachyon*` files, do you want me to modify all of them as you suggested ( `$bin` to `${BIN}` ), to make a standard under those shells ?

Comment (jsimsa) [inline bin/tachyon-mount.sh]: I would do that in a separate PR. It would be great to apply the Google shell style (https://google.github.io/styleguide/shell.xml) as a follow if that is something that interests you.

Comment (caadecarvalho) [inline bin/tachyon-mount.sh]: Sure I'll apply. Thanks for the doc. I'll fix this code as you suggested and create a new [SMALLFIX] PR for the improvements on the other files. Regards.

**Comment by caadecarvalho** [comment]
Jenkins, test this please.

**Comment by caadecarvalho** [comment]
Jenkins, test this please.

**Comment by uronce-cc** [comment]
Thanks for the PR! This solution looks better :+1:

**Comment by uronce-cc** [comment]
LGTM

**Comment by caadecarvalho** [comment]
And I appreciate all your comments and suggestions!

**Comment by jsimsa** [comment]
LGTM, @caadecarvalho thanks for the PR and iterations

**Comment by caadecarvalho** [comment]
I'm the one that must thank you for the knowledge shared!
